cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/batch/MapReduceClassLoader.java

```diff
@@ -140,6 +142,10 @@ public MapReduceTaskContextProvider getTaskContextProvider() {
     LoggingContext loggingContext = createMapReduceLoggingContext();
     LoggingContextAccessor.setLoggingContext(loggingContext);
 
+    // Redirect stdout and stderr for mapreduce to logger
+    System.setOut(new PrintStream(RedirectedPrintStream.createRedirectedOutStream(LOG, System.out), true));
```

Comment:
@chtyim Reason for doing this in MapReduceClassLoader is: DistributedMapReduceTaskContextProvider uses KafkaLogAppender, which is needed to publish stdout/stderr to kafka/UI. Otherwise, if we do it in MapReduceContainerLauncher, then stdout/stderr will only be logged to program.log and not to kafka/UI.